stan/math/rev/functor/algebra_solver_powell.hpp

```diff
@@ -21,57 +22,77 @@ namespace math {
  * function theorem. The call to Jacobian() occurs outside the call to
  * chain() -- this prevents malloc issues.
  */
-template <typename Fs, typename F, typename T, typename Fx>
+template <typename Fy, typename T, typename Fx>
 struct algebra_solver_vari : public vari {
-  /** vector of parameters */
-  vari** y_;
   /** number of parameters */
-  int y_size_;
+  const int y_size_;
+  /** value of parameters */
+  const Eigen::Matrix<T, Eigen::Dynamic, 1> y_val_;
```

Comment:
```suggestion
  arena_t<Eigen::Matrix<T, Eigen::Dynamic, 1>> y_val_;
```

That should work to not leak memory.

`vari` classes don't have destructors because it's assume their memory will all be allocated in the arena and can be cleaned up separately.

An `arena_t<S>` allocates in the arena. Also it looks like y might be an autodiff type, in which case I think `y_` would be a better name than `y_val_` (val_ kinda makes me think this is `value_of(y)`)

Reply:
Thanks, this makes sense. Changed!